### PR TITLE
Add more automated code quality checks

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Closes #

#### What has been done to verify that this works as intended?
Verified that no regressions were introduced manually by running on Nexus 5 - Android 6.0

#### Why is this the best possible solution? Were any other approaches considered?
Adds more checks which check if the stream/resources were properly closed or not.

#### Are there any risks to merging this code? If so, what are they?
No. These changes and fixes were done with the help of the spotbugs plugin.

#### Do we need any specific form for testing your changes? If so, please attach one.
No

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew pmd checkstyle lintDebug spotbugsDebug` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java [hunk 1]
```diff
@@ -488,12 +488,10 @@ public static Bitmap getBitmap(String path, BitmapFactory.Options originalOption
...
     public static byte[] read(File file) {
-        byte[] bytes = {};
-        try {
-            bytes = new byte[(int) file.length()];
-            InputStream is = new FileInputStream(file);
+        byte[] bytes = new byte[(int) file.length()];
+        try (InputStream is = new FileInputStream(file)) {
             is.read(bytes);
-            is.close();
+            return bytes;
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java]: I think this return is redundant. And why don't we need close() anymore?

Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java]: We are using try-with-resources, so it auto closes the stream

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java]: Got it! What about `return`?

Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java]: Yes you are right about the return statement. I'll remove it

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 Please review again

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**